src/sentry/static/sentry/app/components/smartSearchBar/index.tsx

```diff
@@ -94,23 +99,45 @@ const getDropdownElementStyles = (p: {showBelowMediaQuery: number; last?: boolea
   &,
   &:hover,
   &:focus {
-    border-bottom: ${p.last ? null : `1px solid ${theme.border}`};
+    border-bottom: ${p.last ? null : `1px solid ${p.theme.border}`};
     border-radius: 0;
   }
 
   &:hover {
-    color: ${theme.blue300};
+    color: ${p.theme.blue300};
   }
   & > svg {
     margin-right: ${space(1)};
   }
 
   ${
     p.showBelowMediaQuery &&
-    getMediaQuery(theme.breakpoints[p.showBelowMediaQuery], 'flex')
+    getMediaQuery(commonTheme.breakpoints[p.showBelowMediaQuery], 'flex')
   }
 `;
 
+const ThemedCreateSavedSearchButton = withTheme(
```

Comment:
🤔    wouldn't it be better to just wrap `SmartSearchBar` with `withTheme`?

Reply:
just read about it in your description, but shouldn't we investigate why we get type errors and try to fix them?

Reply:
I agree entirely. However, both @Zylphrex and I spent several hours investigating this issue and were unable to identify the root cause of the errors on SmartSearchBarContainer. Putting it on SmartSearchBar had a different set of confusing errors which I did not ask @Zylphrex about but seemed equally rabbit-hole to fix, and this PR has already taken too long. I would be thrilled if a React/Reflux expert could explain to me how to do it properly though.

Reply:
Wrapper the container probably causes issues due to the other HoC's, putting it on `SmartSearchBar` should work though

Reply:
I tried this already, but failed twofold. First I tried @Zylphrex's suggested fix: https://gist.github.com/Zylphrex/98f42d8a66d182b36127028f8e6e1984, which failed tests due to them looking for a component specifically called SmartSearchBar. After fiddling with the names: https://gist.github.com/alxu-sentry/132136f5fbe27d0dcb933dcbc5094624, tests continued to fail with: ReactWrapper::state() can only be called on class components, TypeError: Cannot read property 'state' of null, and ReactWrapper::setState() can only be called on class components. Presumably this is caused by withTheme, and could be fixed by instead exporting non-Themed SmartSearchBar, and making the test specify the theme directly, but that seemed much worse than this solution.

Reply:
Ahhh the tests access state... I *believe* `SmartSearchBar` is exported specifically because of this reason, so we could continue to export the unwrapped `SmartSearchBar`

Reply:
Yes, but then the test needs to specify `theme={lightTheme}`.

Reply:
yes, specifying `theme={light Theme}` would be a solution, but another solution would also be to search for the `SearchBar` component. Ex:

` expect(searchBar.find('SmartSearchBar').state().query).toEqual('one ');`

Reply:
Hm, I didn't think of that. Thank you for the suggestion.

For this PR, I'd rather stick with what I have already, unless someone has strong objections.

Reply:
ok maybe we can update it in another PR :)